cmd/fleetctl/apple_mdm.go

```diff
@@ -44,124 +52,137 @@ func appleMDMCommand() *cli.Command {
-				Usage:       "Validity of the SCEP CA certificate in years",
-				Required:    true,
-				Destination: &validityYears,
+			&cli.StringFlag{
+				Name:     "email",
+				Usage:    "The email address to send the signed APNS csr to.",
+				Required: true,
 			},
 			&cli.StringFlag{
-				Name:        "cn",
-				Usage:       "Common name to set in the SCEP CA certificate",
-				Required:    true,
-				Destination: &cn,
+				Name:     "org",
+				Usage:    "The organization requesting the signed APNS csr.",
+				Required: true,
 			},
 			&cli.StringFlag{
-				Name:        "organization",
-				Usage:       "Organization to set in the SCEP CA certificate",
-				Required:    true,
-				Destination: &organization,
+				Name:     "apns-key",
+				Usage:    "The output path for the APNs private key.",
+				Value:    apnsKeyPath,
+				Required: true,
 			},
 			&cli.StringFlag{
-				Name:        "organizational-unit",
-				Usage:       "Organizational unit to set in the SCEP CA certificate",
-				Required:    true,
-				Destination: &organizationalUnit,
+				Name:     "scep-cert",
+				Usage:    "The output path for the SCEP CA certificate.",
+				Value:    scepCACertPath,
+				Required: true,
 			},
 			&cli.StringFlag{
-				Name:        "country",
-				Usage:       "Country to set in the SCEP CA certificate",
-				Required:    true,
-				Destination: &country,
+				Name:     "scep-key",
+				Usage:    "The output path for the SCEP CA private key.",
+				Value:    scepCAKeyPath,
+				Required: true,
 			},
 		},
 		Action: func(c *cli.Context) error {
-			certPEM, keyPEM, err := scep_ca.Create(validityYears, cn, organization, organizationalUnit, country)
+			email := c.String("email")
+			org := c.String("org")
+			apnsKeyPath := c.String("apns-key")
+			scepCACertPath := c.String("scep-cert")
+			scepCAKeyPath := c.String("scep-key")
+
+			fmt.Fprintf(
+				c.App.Writer,
+				`Sending certificate signing request (CSR) for Apple Push Notification service (APNs) to %s...
+Generating APNs key, Simple Certificate Enrollment Protocol (SCEP) certificate, and SCEP key...
+`,
+				email,
+			)
+
+			// create apns csr and send to fleetdm.com
+			apnsCSR, apnsKey, err := apple_mdm.GenerateAPNSCSRKey(email, org)
 			if err != nil {
-				return fmt.Errorf("creating SCEP CA: %w", err)
+				return fmt.Errorf("generate apns csr: %w", err)
 			}
-			const (
-				certPath = "fleet-mdm-apple-scep.crt"
-				keyPath  = "fleet-mdm-apple-scep.key"
-			)
-			if err := os.WriteFile(certPath, certPEM, 0o600); err != nil {
-				return fmt.Errorf("write %s: %w", certPath, err)
+
+			apnsKeyPEM := apple_mdm.EncodePrivateKeyPEM(apnsKey)
+
+			err = os.WriteFile(apnsKeyPath, apnsKeyPEM, 0600)
+			if err != nil {
+				return fmt.Errorf("write private key: %w", err)
 			}
-			if err := os.WriteFile(keyPath, keyPEM, 0o600); err != nil {
-				return fmt.Errorf("write %s: %w", keyPath, err)
+
+			client := fleethttp.NewClient(fleethttp.WithTimeout(10 * time.Second))
```

Comment:
should this be `client := clientFromCLI(c)`? and then `GetSignedAPNSCSR` is a method of `client`. This is the way the other `fleetctl` commands work.

Reply:
We are not using the fleet server API here. The client ends up being used to make a request to fleetdm.com.

Reply:
🤦 sorry about that, the path looks similar to the schema we use and I completely missed the host.

I think this is out of the scope of this PR, but the code that sends the statistics to fleetdm.com is doing exactly this same thing, something we could refactor in the future.